applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -89,12 +101,83 @@ void GeoLinearElasticPlaneStrain2DLaw::CalculatePK2Stress(const Vector& rStrainV
 {
     KRATOS_TRY
 
+    UpdateInternalDeltaStrainVector(rValues.GetStrainVector());
+
     Matrix C;
     this->CalculateElasticMatrix(C, rValues);
-    // Total formulation
-    noalias(rStressVector) = prod(C, rStrainVector);
+
+    // Incremental formulation
+    noalias(mStressVector) = mStressVectorFinalized + prod(C, mDeltaStrainVector);
+
+    SetExternalStressVector(rStressVector);
 
     KRATOS_CATCH("")
 }
 
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalDeltaStrainVector(const Vector& r_strain_vector){
+    mDeltaStrainVector = r_strain_vector - mStrainVectorFinalized;
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalStrainVectorFinalized(const Vector& r_strain_vector) {
+    this->SetInternalStrainVector(r_strain_vector);
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetExternalStressVector(Vector& r_stress_vector) const{
+    r_stress_vector = mStressVector;
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetInternalStressVector(const Vector& r_stress_vector){
+    mStressVectorFinalized = r_stress_vector;
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetInternalStrainVector(const Vector& r_strain_vector){
+    mStrainVectorFinalized = r_strain_vector;
+}
+void GeoLinearElasticPlaneStrain2DLaw::InitializeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    KRATOS_TRY
+    if (!mIsModelInitialized)
+    {
+        // stress and strain vectors must be initialized:
+        const Vector& r_stress_vector = rValues.GetStressVector();
+        const Vector& r_strain_vector = rValues.GetStrainVector();
+
+        SetInternalStressVector(r_stress_vector);
+        SetInternalStrainVector(r_strain_vector);
```

Comment:
What about eliminating the temporary references?
```suggestion
        SetInternalStressVector(rValues.GetStressVector());
        SetInternalStrainVector(rValues.GetStrainVector());
```

Reply:
Done